src/python/pants/engine/internals/parser.py

```diff
@@ -70,17 +70,19 @@ def __call__(self, *args, **kwargs):
                 self._parse_context._storage.add(obj)
                 return obj
 
-        for alias, symbol in symbol_table.table.items():
-            registrar = Registrar(parse_context, alias, object_type=symbol)
-            symbols[alias] = registrar
-            symbols[symbol] = registrar
-
-        if aliases.objects:
-            symbols.update(aliases.objects)
-
-        for alias, object_factory in aliases.context_aware_object_factories.items():
-            symbols[alias] = object_factory(parse_context)
-
+        symbols.update(
+            {
```

Comment:
Why this change? The old code seems fine, and this is more lines and less readable.

Also, the previous code set both the alias and the symbol keys, this appears to only set the alias. Is that deliberate?

Reply:
Yes, it is deliberate to longer set `symbols[symbol] = registrar`. I'm not sure why we were doing that. It's not possible or desirable to have the symbol be a Python type, as the user can only type a string for the symbol.

I personally think the comprehension is better and our style guide recommends comprehensions in general, but I'll revert - not a big deal either way.